docs/engine/Android-Flutter-Shell-Arguments.md

```diff
@@ -0,0 +1,191 @@
+# Setting Flutter Android engine flags
+
+You can set flags for the Flutter engine on Android in two different ways:
+
+- From the command line when launching an app with the Flutter tool
+- Via `AndroidManifest.xml` metadata (static, per-build configuration)
+
+All flags available on Android can be set via the command line **and** via
+manifest metadata. See `src/flutter/shell/common/switches.cc` for
+the list of all supported flags, and see
+`src/flutter/shell/platform/android/io/flutter/embedding/engine/`
+`FlutterShellArgs.java` for the list of flags that can be set for the
+Android shell.
+
+## When to use manifest metadata versus the command line
+
+Use the manifest when:
+
+- You want a fixed, reproducible baseline of engine flags
+    for your app across all launches. This is ideal for CI and for enforcing a
+    consistent configuration for your app.
+- You want to vary flags by build mode or product flavor
+    via manifest merging. For example, place metadata in
+    `src/debug/AndroidManifest.xml`, `src/profile/AndroidManifest.xml`, and
+    `src/release/AndroidManifest.xml` (or per-flavor manifests) to tailor flags
+    per variant.
+
+Use the command line when:
+
+- You want to quickly experiment with a flag for a single run of your app.
+- You need to override a flag that is already set in the manifest temporarily for debugging
+  or testing purposes.
+
+**Note: If a flag is specified both on the command line and in the manifest,
+the command-line value takes precedence at runtime.**
+
+See below for details on using each method.
+
+## How to set engine flags from the command line
+
+When you run a standalone Flutter app with the Flutter tool, engine flags
+can be passed directly and are forwarded to the Android engine. Examples:
+
+```bash
+flutter run -d android \
```

Comment:
For documentation lets remove flags we do not need or use the expanded form (--device-id)